MetaMask/metamask-extension
# Implement hardware connectivity confirmation screens

Designs found here:  https://www.figma.com/file/6Px3A6vVFGy1hYbEUXi343/Hardware-Wallets?node-id=2%3A2020

## ToDo:

- [x] Implement form fields in popover
- [x] Implement popover padding and other styling
- [x] Remove hardcoded debug values
- [x] Implement polling of connectivity status
- [x] Remove LedgerInstructionsField
- [x] Update Ledger keyring library

## Manual Testing
1.  Pull down this PR
2. In `node_modules/@metamask/eth-ledger-bridge-keyring/index.js`, change the `BRIDGE_URL` to `'https://darkwing.github.io/eth-ledger-bridge-keyring'`; that address includes the `gh-pages` portion of this project (https://github.com/MetaMask/eth-ledger-bridge-keyring/pull/147).  Also, comment out the following check for origin:

```
/*
      if (origin !== this._getOrigin()) {
        return false
      }
*/
```

3. Connect a Ledger address, then try to do a basic send on testnet -- you will see the connectivity messaging and functionality!

## Considerations
1.  Dropdown mode vs. full screen mode
2. The different Ledger connectivity options
